aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -291,6 +292,52 @@ Tensor& sum_out(Tensor& result, const Tensor& self, DimnameList dim,
   return at::sum_out(result, self, dimnames_to_positions(self, dim), keepdim, opt_dtype);
 }
 
+Tensor& nansum_out(Tensor& result, const Tensor& self, IntArrayRef dim,
+                       bool keepdim, optional<ScalarType> opt_dtype) {
+
+  auto get_computation_dtype = [](const Tensor& self) {
+    ScalarType src_type = self.scalar_type();
+    if (at::isIntegralType(src_type, /*includeBool=*/true)) {
+      return c10::typeMetaToScalarType(c10::get_default_dtype());
+    }
+    return src_type;
+  };
+
+  ScalarType scalarType =
```

Comment:
How does PyTorch's sum() type promotion?

Reply:
It's similar to this.